lib/puppet/provider/vc_ip_pool/default.rb

```diff
@@ -0,0 +1,127 @@
+require File.join module_lib, 'puppet_x/vmware/mapper'
+
+Puppet::Type.type(:vc_ip_pool).provide(:vc_ip_pool, :parent => Puppet::Provider::Vcenter) do
+  @doc = "Manage vCenter IpPools. http://pubs.vmware.com/vsphere-55/index.jsp#com.vmware.wssdk.apiref.doc/vim.vApp.IpPool.html"
+  
+  def self.map
+    @map ||= PuppetX::VMware::Mapper.new_map('IpPoolMap')
+  end
+
+  def map
+    self.class.map
+  end
+
+  map.leaf_list.each do |leaf|
+    define_method(leaf.prop_name) do
+      value = PuppetX::VMware::Mapper::munge_to_tfsyms.call(
+        PuppetX::VMware::Util::nested_value(config_is_now, leaf.path_is_now)
+      )
+    end
+
+    define_method("#{leaf.prop_name}=".to_sym) do |value|
+      PuppetX::VMware::Util::nested_value_set(config_should, leaf.path_should, value)
+      
+      @flush_required = true
+    end
+  end
+
+  def create
+    @creating = true
+    @create_message ||= []
+    map.leaf_list.each do |leaf|
+      p = leaf.prop_name
+      unless (value = @resource[p]).nil?
+        self.send("#{p}=".to_sym, value)
+        @create_message << "#{leaf.full_name} => #{value.inspect}"
+      end 
+    end
+
+  end
+
+  def create_message
+    @create_message ||= []
+    "created using {#{@create_message.join ", "}}"
+  end
+
+  alias get_network_association network_association
+  def network_association
+    v = get_network_association
+    networks = [] 
+    v.each{|association| networks << {:networkName => association.networkName} } if v.is_a? Array
+    networks
+  end
+
+  alias set_network_association network_association=
+  def network_association= network_list
+    portgroup_associations = []
+    misses_list = []
+    network_list.each do |network|
+      network_name = network[:networkName] || network['networkName']
+      moref = datacenter.networkFolder.children.find {|pg| pg.name == network_name}
+      if moref
+        portgroup_associations << {:network => moref._ref, :networkName => network_name}
+      else
+        misses_list << network_name
+      end
+    end
+    if misses_list.empty?
+      set_network_association portgroup_associations
+    else
+      raise Puppet::Error, "Unable to locate the following networks in datacenter '#{resource[:datacenter]}' : #{misses_list.inspect}"
+    end
+  end
+
+  def exists?
+    ip_pool
+  end
+
+  def destroy
+    ip_pool_manager.DestroyIpPool(:dc => datacenter, :id => ip_pool.id, :force => resource[:force_destroy])
+  end
+
+  def flush
+   Puppet.debug "config_is_now is #{config_is_now.inspect}"
+   Puppet.debug "config_should is #{config_should.inspect}"
+   if @creating
+     ip_pool_manager.CreateIpPool(:dc => datacenter, :pool => config_should)
+   elsif @flush_required  
+     ip_pool_manager.UpdateIpPool(:dc => datacenter, :pool => config_should)
+   end
+  end
+
+  def config_is_now
+    @config_is_now ||= (@creating ? {} : map.annotate_is_now(ip_pool))
+  end
+
+  def config_should
+    if ip_pool
+      @config_should ||= {:id => ip_pool.id}
```

Comment:
I hadn't thought about it. But if config_should was set to config_is_now then isn't it now a pointer to the same memory space as config_is_now which would mean that as config_should is updated so is config_is_now. I am not sure if that would cause conflicts with the getters and setters. I don't think it should as the getter comparison would always happen first.